packages/server/src/requestPipeline.ts

```diff
@@ -489,18 +484,7 @@ export async function processGraphQLRequest<TContext extends BaseContext>(
     await didEncounterErrors(errors);
 
     const response: GraphQLResponse = {
-      errors: formatErrors(
-        errors.map((err) =>
-          err instanceof ApolloError && !errorClass
-            ? err
-            : fromGraphQLError(
-                err,
-                errorClass && {
-                  errorClass,
-                },
-              ),
-        ),
-      ),
+      errors: formatErrors(errors, errorCode),
     };
 
     // Persisted query errors (especially "not found") need to be uncached,
```

Comment:
BTW you might want to rebase, we've done some small refactorings around here.

Reply:
Done, thanks for the headsup.